### PR TITLE
Updates for wire devices and yaml generation

### PR DESCRIPTION
**Pull request by kabanaty**
Updates for wire devices to address issues #168, #169, and #182.  

- wire.py now has methods to scan and abort wire scans
- wire.py has plane-independent methods for setting ranges and plane selection
- generate.py now accounts for PV fields (ex. MOTR.VELO) and generates yaml files properly

### CODE REVIEW

File: lcls_tools/common/devices/area.py
Comment (MattKing06) [inline lcls_tools/common/devices/area.py:1]: Perfect, thanks @kabanaty!

**Review by MattKing06 — APPROVED** [review]
The yaml generation side of things looks good now and that pattern can be followed for other PVs that require fields. Happy to approve this.

Adding the wires to the `Area` class means people can access magnets/wires/screens all in one place now too! Thanks

**Comment by nneveu** [comment]
Closes #168 #169  #182

**Comment by nneveu** [comment]
@kabanaty not sure why the merge to main for this PR broke the unit tests. Could you please take a look? From my end it looked like your branch was passing the checks.

**Pull request merged**